tflint/config.go

```diff
@@ -228,6 +271,8 @@ func loadConfig(file afero.File) (*Config, error) {
 
 	config := EmptyConfig()
 	config.sources = parser.Sources()
+	// Set the flag if this is a JSON config file
+	config.isJSONConfig = strings.HasSuffix(strings.ToLower(file.Name()), ".json")
```

Comment:
Declare the empty config earlier so you can just do `config.isJSONConfig = true` in the `switch` statement rather than repeating the extension check.